lib/smack/ExtractContracts.cpp

```diff
@@ -92,10 +92,12 @@ std::tuple<BlockList, LoopMap> splitContractBlocks(Function &F, LoopInfo &LI) {
                           << "\n");
             BasicBlock *B = CI->getParent();
             auto NewBB = B->splitBasicBlock(++CI->getIterator());
-            LI.getLoopFor(B)->addBasicBlockToLoop(NewBB, LI);
+            auto L = LI.getLoopFor(B);
+            if (L)
```

Comment:
Should this be an assertion or an if condition? Is the code going to work properly if L is null?

Reply:
It should be an if condition because `B` may not be in any loop.

@michael-emmi could you confirm this?

Reply:
I double-checked this. @shaobo-he is right and I think this is the correct fix.